internal/controller/compaction/reconciler.go

```diff
@@ -278,26 +395,36 @@ func (r *Reconciler) triggerFullSnapshotAndUpdateStatus(ctx context.Context, log
 	if fullSnapErr != nil {
 		latestCondition.Status = druidv1alpha1.ConditionFalse
 		latestCondition.Reason = druidv1alpha1.FullSnapshotFailureReason
-		latestCondition.Message = fmt.Sprintf("Error while triggering full snapshot for etcd %s/%s: %v.\nCompaction will be retried", etcd.Namespace, etcd.Name, fullSnapErr)
+		latestCondition.Message = fmt.Sprintf("Error while triggering full snapshot for etcd %s/%s: %v ,Compaction will be retried", etcd.Namespace, etcd.Name, fullSnapErr)
 	} else {
 		latestCondition.Status = druidv1alpha1.ConditionTrue
 		latestCondition.Reason = druidv1alpha1.FullSnapshotSuccessReason
 		latestCondition.Message = fmt.Sprintf("Full snapshot taken successfully for etcd %s/%s", etcd.Namespace, etcd.Name)
 	}
 	etcdStatusUpdateErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		logger.Info("Updating etcd status condition for full snapshot compaction", "namespace", etcd.Namespace, "name", etcd.Name,
+			"conditionType", latestCondition.Type, "status", latestCondition.Status, "reason", latestCondition.Reason, "message", latestCondition.Message)
+		// Fetch the latest etcd resource to avoid conflict errors
 		latestEtcd := &druidv1alpha1.Etcd{}
 		if err := r.Get(ctx, types.NamespacedName{Namespace: etcd.Namespace, Name: etcd.Name}, latestEtcd); err != nil {
 			return fmt.Errorf("error while fetching etcd %s/%s: %w", etcd.Namespace, etcd.Name, err)
 		}
 		return r.updateCompactionJobEtcdStatusCondition(ctx, latestEtcd, latestCondition)
 	})
+	// If the etcd status update was successful, we will wait for the condition to be reflected in the cache.
+	if etcdStatusUpdateErr == nil {
+		if err := r.waitForEtcdStatusConditionUpdate(ctx, logger, etcd.Namespace, etcd.Name, latestCondition); err != nil {
+			// Don't fail the reconciliation - this is just a cache consistency check
```

Comment:
So if the check fails to confirm the cache is updated within the given timeout period, then we simply continue with the rest of the flow?